lib/Runtime/ByteCode/ByteCodeGenerator.cpp

```diff
@@ -3168,6 +3178,16 @@ void VisitNestedScopes(ParseNode* pnodeScopeList, ParseNode* pnodeParent, ByteCo
                 // Push the param scope
                 byteCodeGenerator->PushScope(paramScope);
 
+                /*
+                We need this call to happen *before* visiting the param scope. If we have a split scope,
+                and this function has parameters that default to a lambda which uses the implicit "arguments" inside
+                of eval, we need this function's argumentsSymbol to be set already, so that we can mark it as
+                non-locally referenced inside of PostVisitFunction.
+
+                See Github issue #1209 for more.
+                */
+                AddVarsToScope(pnodeScope->sxFnc.pnodeVars, byteCodeGenerator);
```

Comment:
This works because `AddVarsToScope` explicitly adds vars to the current function's *body* scope. It uses `ByteCodeGenerator::AddSymbolToFunctionScope`, which in turn calls `ByteCodeGenerator:AddSymbolToScope` with `scope = currentScope->GetFunc()->GetBodyScope()`.